src/Umbraco.Web.UI.Client/src/common/directives/components/tags/umbtagseditor.directive.js

```diff
@@ -165,9 +165,23 @@
         function configureViewModel(isInitLoad) {
             if (vm.value) {
                 if (Utilities.isString(vm.value) && vm.value.length > 0) {
-                    if (vm.config.storageType === "Json") {
+
+                    let parsedJson = undefined;
+                    let isValidJson = true;
+                    if (vm.config.storageType === "Json"){
```

Comment:
It could use the `detectIsJson()` string extension method before trying parsing the json as a few other places in codebase like here:
https://github.com/umbraco/Umbraco-CMS/blob/511ee96c9e0c54480b5350fa0bcabe693bdc1e0a/src/Umbraco.Web.UI.Client/src/common/services/tinymce.service.js#L451